test/integration/model/sync.test.js

```diff
@@ -403,6 +405,33 @@ describe(Support.getTestDialectTeaser('Model'), () => {
           expect(results.filter(r => r.name === 'wow_my_index')).to.have.length(1);
         }
       });
+
+      it('should remove old foreign keys when sync with {alter: true} if custom schema', async function () {
+        if (current.dialect.supports.schemas) {
+          let schema = 'test';
+          if (['mysql', 'mariadb'].includes(current.dialect.name)) {
+            schema = this.sequelize.config.database;
+          } else {
+            await this.sequelize.createSchema(schema);
+          }
+
+          const User = this.sequelize.define('User', {
+            user_id: {
+              type: DataTypes.STRING,
+            },
+          }, { schema });
+          const BelongsToUser = this.sequelize.define('BelongsToUser', {
+            email: {
+              type: DataTypes.STRING,
+            },
+          }, { schema });
+          BelongsToUser.belongsTo(User, { foreignKey: { targetKey: 'user_id', allowNull: false } });
```

Comment:
There is no `targetKey` option in `foreignKey`. If you want to change the targetKey (which defaults to the primary key), it goes on the same level as `foreignKey`